workspace-server/src/main/java/gov/nasa/jpl/aerie/workspace/server/WorkspaceBindings.java

```
@@ -144,6 +172,37 @@ private JWTService.UserSession authorize(Context context) {
     }
   }
 
+  /**
+   * Check that the request meets the permissions to perform the given action on the workspace.
+   * If it does not, format the context into an appropriate error state.
+   * @return true, if the user passes the permissions check. false otherwise
+   */
+  private boolean checkPermissions(Context context, int workspaceId, WorkspaceAction action) {
```

No action needed, but i wonder if a better pattern here could be to `throw` on failed permission checks rather than mutating context and returning a boolean. That would centralize error handling, keep this helper more pure, and avoid callers needing to remember to `return` after the `false` case.

---

I made this method a `boolean` that mutates the context to avoid having duplicate `catch` blocks everywhere that calls the method, but I suppose having it throw an exception that contains the status code and formatted error would work (though that wouldn't necessarily remove the need for a return statement, unless you're proposing to have none of the endpoints catch the new exception)

---

As discussed, I'm going to make a separate small PR that puts that exception into the PermissionsService and updates `MerlinBindings`, `SchedulerBindings`, and `WorkspaceBindings` accordingly.

Edit: [here's the ticket for that work](https://github.com/NASA-AMMOS/aerie/issues/1771)